cggmp21/src/utils.rs

```diff
@@ -215,6 +178,47 @@ pub fn generate_blum_prime(rng: &mut impl rand_core::RngCore, bits_size: u32) ->
     }
 }
 
+/// Takes safe primes p, q, generates and outputs [`PedersenParams`], `phi`, `lambda`
+pub fn generate_pedersen_params(
+    rng: &mut impl rand_core::CryptoRngCore,
+    p: Integer,
+    q: Integer,
+) -> Result<(crate::key_share::PedersenParams, Integer, Integer), GenPedersenError> {
+    use paillier_zk::IntegerExt;
+    use rug::Complete;
+
+    let crt = paillier_zk::fast_paillier::utils::CrtExp::build_n(&p, &q)
+        .ok_or(GenPedersenError::BuildCrt)?;
+    let N = (&p * &q).complete();
+    let phi_N = (&p - 1u8).complete() * (&q - 1u8).complete();
+
+    let r = Integer::gen_invertible(&N, rng);
+    let lambda = Integer::random_below(phi_N.clone() >> 2, &mut external_rand(rng));
+
+    let t = r.square().modulo(&N);
+    let s = crt
+        .exp(&t, &crt.prepare_exponent(&lambda))
```

Comment:
`s = t.exp(λ) mod N` is done by `.exp(&t, &crt.prepare_exponent(&lambda))`.

but `prepare_exponent(..)` is using `phi_n` and not `N` as mentioned in 4.1.1.

Reply:
Added a comment in the spec (best way I found to mention it)
<img width="760" alt="image" src="https://github.com/user-attachments/assets/b7223dc2-ae91-4ac4-ba1d-d6b80d06b18f" />